### ISSUE TITLE
call client.HTTP.NetInfo will get EOF

### ISSUE THREAD

**Issue by yulidai** [tendermint/tendermint#2963]
**Background introduction**
1、Using `https://github.com/certusone/testnet_deploy` to deploy testnet success, and rpc is working fine. i can visit it by browser and get json response
2、Using `https://github.com/certusone/chain_exporter` to monitor by rpc，but failed

**Question description**
After reading `chain_exporter` code, i found the error is thrown by calling `client.HTTP.NetInfo()`. The code is almost the same with: 
```
tClient := client.NewHTTP(url, "/websocket")
netInfo, err := tClient.NetInfo()
if err != nil {
	fmt.Printf("error parsing netData for %s: %v\n", name, err)
}
```
i will get `error parsing netData for **.**.**.**:26657: NetInfo: Post https://**.**.**.**:26657: EOF` after execute it.

**Supplementary information**
other info can see this issue `https://github.com/certusone/chain_exporter/issues/2`

I dont know why `tClient.NetInfo()` will return error even though  rpc is good and network is unobstructed.

Any help will be greatly appreciated :)

**Comment by melekes** [comment]
Are there any errors in the log prior to EOF?

**Comment by yulidai** [comment]
@melekes Sorry about that, it's because i used proxy for speeding network...

But after closing the proxy, i still get error but diffrent:
`NetInfo: Error unmarshalling rpc response result: json: cannot unmarshal object into Go value of type []json.RawMessage`.

As a contrast, if i visit `http://gaia-seed1.interblock.io:26657` will get correct response(although peers is empty array)

our tendermint version is 26.04

**Comment by melekes** [comment]
```
tm-bench -T 500 -r 1000 -c 5 localhost:26657
Status: Error unmarshalling rpc response result: json: cannot unmarshal object into Go value of type []json.RawMessage
```

tm-bench throws the same error.

**Comment by yulidai** [comment]
The problem is the outdated version of the tendermint dependency(0.24.0)

solved after upgrate to `0.27.0-dev1`
